airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/availability_strategy.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+import logging
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple
+
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.availability_strategy import AvailabilityStrategy
+from deprecated.classic import deprecated
+
+
+@deprecated("This class is experimental. Use at your own risk.")
+class AbstractAvailabilityStrategy(ABC):
+    """
+    AbstractAvailabilityStrategy is an experimental interface developed as part of the Concurrent CDK.
+    This interface is not yet stable and may change in the future. Use at your own risk.
+
+    Why create a new interface instead of using the existing AvailabilityStrategy?
+    The existing AvailabilityStrategy is tightly coupled with Stream and Source, which yields to circular dependencies and makes it difficult to move away from the Stream interface to AbstractStream.
+    """
+
+    @abstractmethod
+    def check_availability(self, logger: logging.Logger) -> Tuple[bool, Optional[str]]:
```

Comment:
this method might require additional parameters as we implement a featureful implementation, but it should not depend on `Stream` or `Source`

Reply:
IMHO, worse considering to use namedtuples here